### ISSUE TITLE
[Req] Test templates changes between releases

### ISSUE THREAD

**Issue by san7ket** [SatelliteQE/satellite6-upgrade#80]
It would be great to have automated test for verifying that an existing template with content "a" e.g. in 6.2 is automatically updated to content "b" after update to 6.3. The most important template would be the kickstart one which I'm I think changed between versions. Maybe it would be best to verify that after update, the template in database is 1:1 to the version that is on filesystem. Templates can be found at ~foreman/app/views/unattended/provisioning_templates and /opt/theforeman/tfm/root/usr/share/gems/gems/foreman_theme_satellite-*/app/views/foreman/unattended. The later directory contains the kickstart template. The same behavior should be observable for partition tables. It would be also to good to verify that templates that are shipped in 6.3 by default are locked. Note that whenever foreman-rake db:seed is run (usually from satellite installer), the templates in DB should be updated to versions on filesystem.

@san7ket is there such test or could it be added to the upgrade suite? @rplevka could you please add the right flag to this issue please?

**Comment by rplevka** [comment]
I don't think this got to exist here, we might have kept the robottelo issue opened.
Since we're about to compare `db` vs `fs`,  this test will make sense even in regular (non-upgraded environment.

**Comment by sghai** [comment]
@rplevka cool, please feel free to close this issue and linked it up w/ the one opened on robottelo.

**Comment by rplevka** [comment]
@san7ket, do you agree?

**Comment by san7ket** [comment]
yea, I think so, but still we can also have this here , to check the content changes were done to the templates.
Currently, we have tests to check the template entity exists (pre/post comparison), but we do not check the content changes that were done to templates

**Comment by san7ket** [comment]
@rplevka If we are going to have this in robottelo, please feel free to close this issue here.

**Comment by rplevka** [comment]
Or, maybe we should come up with the tests that actually use the templates - this way we wouldn't test whether some templates got changed, but whether they continue to work instead.
I know this would require a number of tests, but maybe we can cover most of them by few end-to-ends

**Comment by jyejare** [comment]
@rplevka , That will be a huge effort to put. Better we can ask dev's to fetch template contents either from API or CLI and compare that data with Upgrade Existence Tests.

@sghai , @san7ket @ntkathole  Thoughts ?

**Comment by sghai** [comment]
A Template contains many statements apart from plain os provisioning, so with provisioning, we may not be able to identify gaps and moreover covering all possible scenarios required more efforts. So existence tests would be a good way to go, however,lets see how much efforts needed for existence tests as well.
@jyejare : could you please check efforts required to automate via existence tests ?
